### PR TITLE
Math: Added linear_interpolate to misc math

### LINKED ISSUE #1880: Actuator: Transplant MixerCurve from actuator.c into math libs; add unit tests
- Pending #1859 
- Consider renaming to something like linear_interpolate or linear_interpolate_curve
- Determine unit test conditions

### PR DESCRIPTION
**Pull request by ekaszubski**
Fixes #1880 
- Added unit tests in misc_math/unittest.cpp
- Updated references in actuator.c

### CODE REVIEW

**Comment by peabody124** [comment]
Love it!

File: flight/tests/misc_math/unittest.cpp [hunk 1]
```diff
@@ -237,3 +237,63 @@ TEST_F(CircularModulusDeg, SweepError) {
...
+
+  // 21 points in range
+  for (size_t i = 0; i <= 20; ++i) {
+    float const input = i * 0.05f;
+    EXPECT_EQ(input, linear_interpolate(input, curve, curve_numpts, range_min, range_max));
+  }
+
+  // 10 points below min range
+  for (size_t i = 1; i <= 10; ++i) {
+    float const input = range_min - i * 0.1f;
```
Comment (kubark42) [inline flight/tests/misc_math/unittest.cpp]: This isn't really an issue because you do it in the above tests, but it doesn't explicitly test the behavior at the boundary. If you do another commit, it might be good to test that ```range_max``` maps to ```range_max```, in case the first unit test gets changed.

**Comment by kubark42** [comment]
@ekaszubski Cool stuff, thanks.

Should there also be a test for non-monotonic curves? I don't see it makes a big difference, since the curve is anyway monotonic inside a bin.

File: flight/Libraries/math/misc_math.c [hunk 2]
```diff
@@ -255,6 +255,43 @@ void cubic_deadband_setup(float w, float b, float *m, float *r)
 	*r = *m * powf(w, 3) + b * w;
 }
 
+/**
+ * Perform a linear interpolation over N points
```
Comment (kubark42) [inline flight/Libraries/math/misc_math.c]: My apologies if I'm being overly pedantic, but I wonder if we could do better for the naming here. It's only because that since its mapping is constrained to [0,-1], it's not a full linear interpolation.

Comment (ekaszubski) [inline flight/Libraries/math/misc_math.c]: @kubark42 the mapping is actually constrained to [input_min, input_max] and the number of points is adjustable, so I believe it is a full linear interpolation

**Comment by ekaszubski** [comment]
@kubark42 Agreed; I started working on a cubic test at least for CollectiveCurve but it's not completed yet

File: flight/tests/misc_math/unittest.cpp (hunk 1 above)
Comment (ekaszubski) [inline flight/tests/misc_math/unittest.cpp]: @kubark42 By "not explicitly" do you mean there isn't a named TEST_F for testing the boundaries? Because they are covered in the 21-point test (first and last point are the min and max bounds).

File: flight/Libraries/math/misc_math.c (hunk 2 above)
Comment (kubark42) [inline flight/Libraries/math/misc_math.c]: The output range is constrained to [0,1], right? So I couldn't provide a set of ```x``` values with corresponding ```y``` values, and then get the ```y_interpolated``` vs ```x_intermediate``` values.

Comment (ekaszubski) [inline flight/Libraries/math/misc_math.c]: @kubark42 There are no constraints on the inputs or outputs; you can set the input range to -50,50 and set the curve points to [-5,-2.5,-,2.5,5] or whatever you want; the min/max output will be defined by the curve points. The [0,1] thing is only used to select the curve point index to use for interpolation, but the point at that index has no constraints on its range.

**Comment by mlyle** [comment]
Not for action, just a remark:  It might be nice to have this https://en.wikipedia.org/wiki/Monotone_cubic_interpolation eventually

**Comment by mlyle** [comment]
I am marking PRs to merge in the first pass after a release branch is created. They are not for merge now.

**Comment by mlyle** [comment]
Argh picked up conflicts on this one.

**Comment by ekaszubski** [comment]
@mlyle Will fix this evening

**Comment by mlyle** [comment]
Cool.  Sorry for the conflicts, etc.. this is the downside of closing the merge window for awhile.

**Comment by ekaszubski** [comment]
@mlyle It's all good :)

**Comment by mlyle** [comment]
And... merged
